src/lib/broadcast.jl

```diff
@@ -239,7 +238,13 @@ function dual_function(f::F) where F
   end
 end
 
-@inline function broadcast_forward(f, args::Vararg{Any,N}) where N
+@inline function broadcast_forward(f::F, args::Vararg{Any,N}) where {F,N}
+  Base.issingletontype(F) || error("""Zygote's dual number broadcasting (as used on GPU arrays) cannot handle this function.
+      typeof(f) = $(F)""")
```

Comment:
Maybe this should be a warning like #1224. We don't know that `f` ought to have a gradient, but it might.